### ISSUE TITLE
Rewrite ICE

### ISSUE THREAD

**Issue by PeterTillema** [PeterTillema/ICE#126]
Title says it all. It completely needs to be rewritten and refined, with a much better syntax and source code.

**Comment by Overload02** [comment]
Do you really need to rewrite it?! Just use the same src and add onto it. 😄

**Comment by PeterTillema** [comment]
> Do you really need to rewrite it?! Just use the same src and add onto it. 😄

Yes. The current source code is garbage and it needs a totally different structure.
